### PR TITLE
Feature/resolverbugfix

### PR DESCRIPTION
**Pull request by srinithi19**

### CODE REVIEW

File: server/schemas/resolvers.js [hunk 1]
```diff
@@ -9,19 +9,25 @@ const resolvers = {
         getTicketById: async (parent, {ticketId, userType}) => {
 
             if (userType == "Agent") {
-                return Ticket.findOne({ticketId}).populate('comments');
+                return Ticket.findOne({_id:ticketId}).populate('comments').populate('users');
```
Comment (megellman) [inline server/schemas/resolvers.js:12]: why are we populating users here? aren't we getting all the user info we need from the comment? is there another circumstance where we would need to include users?

File: server/schemas/resolvers.js [hunk 2]
```diff
@@ -9,19 +9,25 @@ const resolvers = {
...
-                return Ticket.findOne({ticketId}).populate('comments','-note');
+                return Ticket.findOne({_id:ticketId}).populate('comments', '-notes').populate('users');
             }
       },
 
       // get Tickets by userId and Status
       getTicketsByUserId: async (parent, {userId , status}) => {
-        return Ticket.find({
+        if(status)
+            return Ticket.find({
```
Comment (megellman) [inline server/schemas/resolvers.js:22]: the if statement needs brackets, like this: 
 if(status){
            return Ticket.find({
            users : userId,
            status : status
}
        });

File: server/schemas/typeDefs.js [hunk 3]
```diff
@@ -3,34 +3,34 @@ const { gql } = require('apollo-server-express');
 const typeDefs = gql`
     type User {
         _id: ID!
-        firstName: String!
-        lastName: String!
-        password: String!
+        firstName: String
```
Comment (megellman) [inline server/schemas/typeDefs.js:6]: I think these should have ! included. We can discuss this more in our meeting but I don't think this is the currect solution to the problem you mentioned in slack --- "returns error saying cannot return null for non - nullable field". I think that issue had more to do with the queries themselves and what they were returning

**Review by megellman — CHANGES_REQUESTED** [review]

File: server/schemas/resolvers.js (hunk 1 above)
Comment (srinithi19) [reply server/schemas/resolvers.js:12]: if we are not populating users, it displays just their ID. to display their date, we need to populate users

File: server/schemas/resolvers.js (hunk 2 above)
Comment (srinithi19) [reply server/schemas/resolvers.js:22]: Does not matter for single line of code inside if else

File: server/schemas/typeDefs.js (hunk 3 above)
Comment (srinithi19) [reply server/schemas/typeDefs.js:6]: Nope, it means "String! means that the field is non-nullable, meaning that the GraphQL service promises to always give you a value when you query this field. In the type language, we'll represent those with an exclamation mark."

https://graphql.org/learn/schema/

**Review by rmoscoe — APPROVED** [review]

**Review by megellman — APPROVED** [review]

**Pull request merged**